clang/lib/AST/ByteCode/Compiler.cpp

```diff
@@ -5189,6 +5197,12 @@ template <class Emitter> bool Compiler<Emitter>::visitIfStmt(const IfStmt *IS) {
       return false;
   }
 
+  if (auto *DD =
+          dyn_cast_if_present<DecompositionDecl>(IS->getConditionVariable());
+      DD && DD->isDecisionVariable())
```

Comment:
When is `DD->isDecisionVariable()` false? Should we assert instead?
Ditto elsewhere